soda/contracts/docs/02_verifying_a_contract_in_python/02_verifying_a_contract_with_the_api.md

```diff
@@ -0,0 +1,183 @@
+# Contract API
+
+This section explains the Python API to verify a contract.
+
+The contract object is created from a YAML file (or string).
+
+Contract verification performed executed on a SQL connection to a warehouse.
+
+
+## Requires
+
+* A virtual environment with `soda-core-contracts` installed.
+  See [Setting up a soda contracts virtual environment](./01_setting_up_a_python_virtual_environment_for_soda_contracts)
+
+## The API
+
+First we start with a complete example that you can copy-and-paste.  Then we explain the sections in there one by one.
+
+```python
+from soda.contracts.connection import Connection
+from soda.contracts.contract import Contract, ContractResult
+
+try:
+    with Connection.from_yaml_file("./postgres_localhost_dev.scn.yml") as connection:
+        contract: Contract = Contract.from_yaml_file("./customers.sdc.yml")
+        contract_result: ContractResult = contract.verify(connection)
+except SodaException as e:
+    # make the orchestration job fail and report all problems
+```
+
+## Creating a connection
+
+A contract is verified over a connection to a SQL engine. Internally the implementation is based on
+a Python DBAPI connection.
+
+A connection can be created based on a YAML connection configuration file, a YAML string or a dict.
+Use one of the `Connection.from_*` methods to create a connection.
+
+```python
+with Connection.from_yaml_file("./postgres.scn.yml") as connection:
+    # Do stuff with connection
+```
+
+### Connection YAML configuration files
+
+It's good practice to use environment variables in your YAML files to protect your credentials.
+Use syntax `${POSTGRES_PASSWORD}` for resolving environment variables.  Variable resolving is
+case sensitive. We recommend using all upper case and underscores for naming environment variables.
+
+For connection configuration YAML files, we recommend to use extension `scn.yml` (Soda ConnectioN) because
```

Comment:
the `scn` part seems a bit cryptic, like something no one will remember and will always have to look it up. How about a verbose option like `.soda.connection.yml` and `.soda.contract.yml` for 100% clarity? We could easily support both but I would surely pick the verbose one myself

Reply:
I agree with the clarity on the long form.  Still a bit unsure as it may be too long.  Maybe ....contract.yml and ...is enough?  I ll give it some thought.

Reply:
that's true, I think contract.yml is enough